thunder/torch/__init__.py

```diff
@@ -3364,6 +3364,20 @@ def matmul(a: TensorLike, b: TensorLike, /) -> TensorLike:
     if a.ndim == 1 or b.ndim == 1:
         return prims.matmul(a, b)
 
+    # Case nd @ 2d --> reduce to a 2d gemm
+    if a.ndim > 2 and b.ndim == 2:
+        a_batch_dims = a.shape[:-2]
+
+        # a -> a_2d by flattening batch dims with the row space
+        a_2d = a.reshape(-1, a.shape[-1])
```

Comment:
This is OK because we know that PyTorch would do it, too, right?

Reply:
PyTorch will, our prim and lowering otherwise will not.

Reply:
Yeah, just that we don't introduce more copies when things are non-contiguous. Let's have it.